static/src/js/components/PermissionForm/PermissionForm.jsx

```diff
@@ -198,6 +198,14 @@ const PermissionForm = ({ selectedCollectionsPageSize }) => {
     }
   })
 
+  useEffect(() => {
+    if (providerId) {
+      const newUiSchema = cloneDeep(uiSchema)
+      newUiSchema.collectionSelection.selectedCollections['ui:providerId'] = providerId
```

Comment:
What is the purpose of modifying the UI schema, I'm not clear on the intentions.

Reply:
In order to pass the providerId to the collection selector field, it needs to go through the uischema.   This is so we can filter that left hand side (available collections) by the providerId.

Reply:
I'd like to see a demo of this I'm not following. The ui schema is a ui facade for the data, the data should provide this type of information, but I could definitely be wrong.

Reply:
Sure, there are multiple ways to get the provider id into the widget, but I can show you at least my ratio for why I choose this path.

Reply:
I removed this logic of passing providerId through uischema in favor or just grabbing it from `AppContext`.